src/swap.ts

```diff
@@ -82,76 +156,163 @@ export class Swap extends Core {
     );
 
-  const outputRFound = outputFoundInTransaction(
+  // check if the output if found in the transaction
+  const outputRFound: boolean = outputFoundInTransaction(
     decodedFromRequest.transaction.outs,
     msgRequest.getAmountR(),
-    msgRequest.getAssetR()
+    msgRequest.getAssetR(),
+    blindKeysMap(msgRequest.getOutputBlindingKeyMap())
   );
+
   if (!outputRFound)
     throw new Error(
       'Either SwapRequest.amount_r or SwapRequest.asset_r do not match the provided psbt'
     );
 
+  // msg accept
   if (msgAccept) {
+    // decode the tx and check the msg's ids
     const decodedFromAccept = decodePsbt(msgAccept.getTransaction());
     if (msgRequest.getId() !== msgAccept.getRequestId())
       throw new Error(
         'SwapRequest.id and SwapAccept.request_id are not the same'
       );
 
+    // check the amount of utxos.
     const totalR = countUtxos(
       decodedFromAccept.psbt.data.inputs,
-      msgRequest.getAssetR()
+      msgRequest.getAssetR(),
+      blindKeysMap(msgAccept.getInputBlindingKeyMap())
     );
+
     if (totalR < msgRequest.getAmountR())
       throw new Error(
         'Cumulative utxos count is not enough to cover SwapRequest.amount_r'
       );
 
+    // check if there is an output found in the transaction.
     const outputPFound = outputFoundInTransaction(
       decodedFromAccept.transaction.outs,
       msgRequest.getAmountP(),
-      msgRequest.getAssetP()
+      msgRequest.getAssetP(),
+      blindKeysMap(msgAccept.getOutputBlindingKeyMap())
     );
+
     if (!outputPFound)
       throw new Error(
         'Either SwapRequest.amount_p or SwapRequest.asset_p do not match the provided psbt'
       );
   }
 }
 
+/**
+ * find an output in outputs corresponding to value and asset. Provide outputBlindKeys if output are blinded.
+ * @param outputs the outputs to search in.
+ * @param value value of the output.
+ * @param asset hex encoded asset of the output.
+ * @param outputBlindKeys optional, only if blinded outputs. Blinding keys map (scriptPukKey -> blindingKey).
+ */
 function outputFoundInTransaction(
-  outputs: Array<any>,
+  outputs: Array<Output>,
   value: number,
-  asset: string
-) {
-  const found = outputs.find(
-    (o: any) => toNumber(o.value) === value && toAssetHash(o.asset) === asset
-  );
-
-  return found !== undefined;
+  asset: string,
+  outputBlindKeys: BlindKeysMap = {}
+): boolean {
+  return outputs.some((o: Output) => {
+    // unblind first if confidential ouput
+    if (isConfidentialOutput(o)) {
+      const blindKey: Buffer = outputBlindKeys[o.script.toString('hex')];
+      // if no blinding keys for the confidential ouput --> return false
+      if (blindKey === undefined) return false;
+      try {
+        const { value: unblindValue, asset: unblindAsset } = unblindOutput(
+          o,
+          blindKey
+        );
+        // check unblind value and unblind asset
+        return parseInt(unblindValue, 10) === value && unblindAsset === asset;
```

Comment:
you should use confidentialValueToSatoshi

Reply:
The liquidjs-lib `unblindOutput` function already translate confidential value to string: https://github.com/provable-things/liquidjs-lib/blob/master/ts_src/confidential.ts#L62